File: javascript/leetcode/isomorphic-strings.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isIsomorphic = function(s, t) {
    return encodeIsomorphic(s) == encodeIsomorphic(t)
};

var encodeIsomorphic = function(str) {
    sH = {}
    sR = ''
    sC = 0
    for (var i = 0; i < str.length; i++) {
        if (sH[str[i]] != null) {
            sR += sH[str[i]]
        } else {
            sH[str[i]] = sC
            sR += sH[str[i]]
            sC++
        }
    }

    return sR
} 

File: python/leetcode/isomorphic-strings.py
class Solution:
    def encodeIsomorphic(self, s):
        sH = {}
        sR = ''
        sC = 0
        for c in s:
            if c in sH:
                sR += str(sH[c])
            else:
                sH[c] = sC
                sR += str(sH[c])
                sC += 1
        return sR

    def isIsomorphic(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        return self.encodeIsomorphic(s) == self.encodeIsomorphic(t)
